### DISCUSSION TITLE
Error: failed to get lock for object

### DISCUSSION THREAD

**Discussion by lukasz-borek** [minio/minio#17746] [Q&A] [answered]
Hi! 

How to troubleshoot errors like that: 

```
Jul 28 09:52:37 minio2-nodehdd1 minio[1440895]: DeploymentID: 11332337-0584-491c-aa41-1c7161d9410a
Jul 28 09:52:37 minio2-nodehdd1 minio[1440895]: Error: failed to get lock for object: ble/2023-07-27T20:51:23Z/ble/ble.orders.s2 bucket:mongobackup-prod arn: (*errors.errorString)
Jul 28 09:52:37 minio2-nodehdd1 minio[1440895]:        3: internal/logger/logger.go:258:logger.LogIf()
Jul 28 09:52:37 minio2-nodehdd1 minio[1440895]:        2: cmd/bucket-replication.go:1009:cmd.replicateObject()
Jul 28 09:52:37 minio2-nodehdd1 minio[1440895]:        1: cmd/bucket-replication.go:1741:cmd.(*ReplicationPool).AddMRFWorker()
```

mcli stat:
```
Name      : mongobackup-prod/ble/2023-07-13T08:26:14Z/ble-rs0/ble_p_p.views.s2
Date      : 2023-07-13 13:12:58 CEST
Size      : 185 GiB
ETag      : 016d7866e8a413d6d8f18a71c1578fa9-9403
VersionID : 1a880b8b-5b73-443a-84fb-4ccd5f56abd6
Type      : file
Metadata  :
  X-Amz-Object-Lock-Mode             : GOVERNANCE
  Content-Type                       : binary/octet-stream
  X-Amz-Object-Lock-Retain-Until-Date: 2023-07-27T08:26:24.803Z
Replication Status: FAILED
```

Size of the object is an issue here? What kind of lock error is referring to? 

Version: 2023-07-21T21:12:44Z
4 x physical nodes, 12 HDD each
Pools:
   1st, Erasure sets: 3, Drives per erasure set: 16

87 TiB Used, 2 Buckets, 12,852 Objects, 13,808 Versions, 955 Delete Markers
48 drives online, 0 drives offline

**Comment by ravindk89** [comment]
The logs are referencing bucket replication. And it looks like you have bucket replication configured here - but for whatever reason, replication could not acquire a lock on that object. 

How is your bucket replication configured, and do you see any other network errors between your primary and DR site?

**Comment by harshavardhana** [comment]
The object here is very large without sufficient bandwidth this will fail, the lock message is due to active uploading that is in progress.

**Comment by lukasz-borek** [comment]
Network is quite solid between sites. iperf show 500MB/s speed. As it's backup app, objects size stars from few KB up to 500GB. 

So lock error is temporary. One replication thread lock the file during transfer and not allow another to jump in? 

Is there a way to see if an object is currently locked?

**Comment by lukasz-borek** [comment, marked as answer]
Closing the discussion. Replication is catching up, lock errors are only occurring in 100GB+ objects and finally get replicated. 

Not sure if you guys want to fix it, if yes i can open an issue.

**Comment by harshavardhana** [comment]
> Closing the discussion. Replication is catching up, lock errors are only occurring in 100GB+ objects and finally get replicated.


// cc @poornas

**Reply by poornas** [reply]
there is no issue to fix here - a replication attempt is in progress and other concurrent attempts will not succeed because of failure to get replication lock. This is expected
